lib/router.js

```diff
@@ -219,6 +219,13 @@ Router.prototype._handleSwaggerPathSpec = function(node, pathspec,
         });
     }
 
+    var security = pathspec.security;
```

Comment:
I think you can actually handle this synchronously, before the loaderPromise is even created. If your concern is about modules adding security restrictions, then I think we can add logic for that above.